core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -130,4 +130,21 @@ class HttpWorkspaceManagerDAO(apiClientProvider: WorkspaceManagerApiClientProvid
   def getCreateAzureRelayResult(workspaceId: UUID, jobControlId: String, accessToken: OAuth2BearerToken): CreateControlledAzureRelayNamespaceResult = {
     getControlledAzureResourceApi(accessToken).getCreateAzureRelayNamespaceResult(workspaceId, jobControlId)
   }
+
+  def createAzureStorageAccount(workspaceId: UUID, region: String, accessToken: OAuth2BearerToken) = {
+    // Storage account names must be unique and 3-24 characters in length, numbers and lowercase letters only.
+    val prefix = workspaceId.toString.substring(0, workspaceId.toString.indexOf("-"))
+    val suffix = workspaceId.toString.substring(workspaceId.toString.lastIndexOf("-") + 1)
+    getControlledAzureResourceApi(accessToken).createAzureStorage(
+      new CreateControlledAzureStorageRequestBody().common(
+        new ControlledResourceCommonFields().name(s"storage-rcf-${workspaceId}").
+          cloningInstructions(CloningInstructionsEnum.NOTHING).
+          accessScope(AccessScope.SHARED_ACCESS).
+          managedBy(ManagedBy.USER)
+      ).azureStorage(
+        new AzureStorageCreationParameters().name(s"sa${prefix}${suffix}").region(region)
```

Comment:
This produces a 22-char name.

Reply:
Would it be overkill to call `.take(24)` on this name to safeguard against changes to the workspace id format? I don't see why we'd change the workspace id format, but it popped into my head.

Reply:
I don't think we will stop using a UUID, but if we did and the length was > 24 chars, failures would occur immediately.